packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -852,23 +861,35 @@ class _ToggleButton extends StatelessWidget {
   /// The button's label, which is usually an [Icon] or a [Text] widget.
   final Widget child;
 
+  Color _resolveColor(Set<MaterialState> states, MaterialStateProperty<Color?> widgetColor,
+  MaterialStateProperty<Color?> themeColor, MaterialStateProperty<Color> defaultColor) {
+  return widgetColor.resolve(states)
+    ?? themeColor.resolve(states)
+    ?? defaultColor.resolve(states);
+  }
+
   @override
   Widget build(BuildContext context) {
     assert(debugCheckHasMaterial(context));
     final Color currentColor;
-    final Color currentFillColor;
     Color? currentFocusColor;
     Color? currentHoverColor;
     Color? currentSplashColor;
     final ThemeData theme = Theme.of(context);
     final ToggleButtonsThemeData toggleButtonsTheme = ToggleButtonsTheme.of(context);
 
+    final Set<MaterialState> states = <MaterialState>{
+        if (selected && onPressed != null) MaterialState.selected,
+        if (onPressed == null) MaterialState.disabled,
+    };
+
+    final Color currentFillColor = _resolveColor(states, _ResolveFillColor(fillColor),
+    _ResolveFillColor(toggleButtonsTheme.fillColor), _DefaultFillColor(theme.colorScheme));
```

Comment:
The _ResolveFillColor class shouldn't be necessary, _DefaultFillColor should capture the defaults.

```dart
 final Color currentFillColor = _resolveColor(states, fillColor, toggleButtonsTheme.fillColor, _DefaultFillColor(theme.colorScheme);
```

Reply:
Still need to reformat this